### PR TITLE
GDB-7730: Integration in jdbc view

### PR DESCRIPTION
**Pull request by boyan-tonchev**
## What
Refactoring "Jdbc create" view.

## Why
We want to integrate "ontotext-yasgui-web-component" to WB.

#How
Replaces the old implementation with new one, because it is not applicable with "ontoext-yasgui-web-component". Implements test to cover the create/update jdbc configuration functionality.

Additional work:
Replaces old sparql view with new one.

### CODE REVIEW

File: src/js/angular/utils/repositories-utils.service.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const modules = [];
+
+angular
+    .module('graphdb.framework.repositories', modules)
+    .factory('RepositoriesUtilService', RepositoriesUtilsService);
+
+RepositoriesUtilsService.$inject = ['RDF4JRepositoriesRestService'];
+
+function RepositoriesUtilsService(RDF4JRepositoriesRestService) {
```
Comment (svilenvelikov) [inline src/js/angular/utils/repositories-utils.service.js:9]: Didn't we untangled the cyclic dependency between the services which enforsed you to write down this redundant utility service?

File: src/css/jdbc.css [hunk 2]
```diff
@@ -1,3 +1,8 @@
 .hidden {
     visibility: hidden;
 }
+
+.yasr_results {
+    max-height: 345px;
```
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply src/css/jdbc.css:6]: I took this value from the old implementation.

File: src/js/angular/jdbc/controllers.js [hunk 3]
```diff
@@ -53,510 +62,630 @@ function JdbcListCtrl($scope, $repositories, JdbcRestService, toastr, ModalServi
...
+        if (!$scope.ontotextYasgui) {
+            $scope.ontotextYasgui = document.querySelector('ontotext-yasgui');
+        }
+        return $scope.ontotextYasgui;
     };
 
-    $scope.containsUnknownColumns = function () {
-        return $scope.currentQuery.columns && $scope.currentQuery.columns
-            .find((el) => el.column_type === 'unknown');
+    const updateQuery = (jdbcConfigurationInfo) => {
```
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply src/js/angular/jdbc/controllers.js:485]: Yes, you are right. I have to rename it. What you think about "updateJdbcConfigurationQuery"?

File: src/js/angular/jdbc/controllers.js [hunk 4]
```diff
@@ -53,510 +62,630 @@ function JdbcListCtrl($scope, $repositories, JdbcRestService, toastr, ModalServi
...
     };
 
-    if ($scope.name) {
-        getJdbcConfiguration($scope.name);
-    } else {
-        setQueryFromTabConfig();
-    }
+    $scope.selectColumnType = (columnName, prevColumnType) => {
+        $scope.setDirty();
+        const column = _.find($scope.jdbcConfigurationInfo.columns, function (column) {
```
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply src/js/angular/jdbc/controllers.js:224]: This is from old implementation. I will change it.

File: src/js/angular/jdbc/controllers.js [hunk 5]
```diff
@@ -53,510 +62,630 @@ function JdbcListCtrl($scope, $repositories, JdbcRestService, toastr, ModalServi
...
+            }).catch(function (e) {
+                toastr.error(getError(e), $translate.instant('jdbc.not.suggest.column.types'));
+            });
     };
 
-    $scope.containsNonLiteralColumnsOnly = function (columns) {
-        return columns && columns.every((el) => !$scope.isLiteral(el.column_type));
+    const getOntotextYasgui = () => {
+        if (!$scope.ontotextYasgui) {
+            $scope.ontotextYasgui = document.querySelector('ontotext-yasgui');
```
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply src/js/angular/jdbc/controllers.js:480]: Hm. I used the reference "$scope.ontotextYasgui", but I changed that with usage of the function "getOntotextYasgui", now is meaningless to stay in the scope. I will remove it.

File: src/js/angular/utils/repositories-utils.service.js (hunk 1 above)
Comment (boyan-tonchev) [reply src/js/angular/utils/repositories-utils.service.js:9]: I will check this.

Comment (boyan-tonchev) [reply src/js/angular/utils/repositories-utils.service.js:9]: Yes, the feature was moved to the repositories service. I just forgot to delete this.

File: src/js/angular/jdbc/controllers.js [hunk 6]
```diff
@@ -53,510 +62,630 @@ function JdbcListCtrl($scope, $repositories, JdbcRestService, toastr, ModalServi
...
+    $scope.sqlTypes = [];
+    $scope.activeTab = 1;
+    $scope.jdbcConfigurationInfo = new JdbcConfigurationInfo();
+    $scope.jdbcConfigurationInfo.jdbcConfigurationName = $location.search().name || '';
+    $scope.saveOrUpdateExecuted = false;
+    $scope.jdbcConfigurationInfo.isNewJdbcConfiguration = !$scope.jdbcConfigurationInfo.jdbcConfigurationName;
+    $scope.language = $languageService.getLanguage();
+    $scope.isDirty = false;
+    $scope.prefixes;
+    $scope.isLoading;
```
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply src/js/angular/jdbc/controllers.js]: Done

File: src/js/angular/jdbc/controllers.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply src/js/angular/jdbc/controllers.js]: Done

File: src/js/angular/jdbc/controllers.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply src/js/angular/jdbc/controllers.js]: done

File: src/js/angular/jdbc/controllers.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply src/js/angular/jdbc/controllers.js]: Done

File: src/js/angular/utils/repositories-utils.service.js (hunk 1 above)
Comment (boyan-tonchev) [reply src/js/angular/utils/repositories-utils.service.js]: Done

File: src/js/services/ontotext-yasgui-web-component.service.js [hunk 7]
```diff
@@ -17,15 +33,15 @@ function OntotextYasguiWebComponentService(MonitoringRestService, RDF4JRepositor
         if (req) {
             const repository = req.url.substring(req.url.lastIndexOf('/') + 1);
             const currentTrackAlias = req.header['X-GraphDB-Track-Alias'];
-            return MonitoringRestService.deleteQuery(currentTrackAlias, repository)
+            return monitoringRestService.deleteQuery(currentTrackAlias, repository)
```
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply src/js/services/ontotext-yasgui-web-component.service.js]: Done

**Comment by boyan-tonchev** [comment]
Jenkins retry build

**Comment by boyan-tonchev** [comment]
Jenkins retry build

**Comment by boyan-tonchev** [comment]
Jenkins retry build

**Comment by boyan-tonchev** [comment]
Jenkins retry build

**Comment by boyan-tonchev** [comment]
Jenkins retry build

**Comment by boyan-tonchev** [comment]
Jenkins retry build

**Comment by boyan-tonchev** [comment]
Jenkins retry build

**Pull request merged**